Lib/test/test_struct.py

```diff
@@ -652,6 +652,17 @@ def test_format_attr(self):
         s2 = struct.Struct(s.format.encode())
         self.assertEqual(s2.format, s.format)
 
+    def test_uninitialized_struct(self):
+        s = struct.Struct.__new__(struct.Struct)
+        with self.assertRaises(ValueError):
+            s.format
+        with self.assertRaises(ValueError):
+            s.size
+        for meth in s.iter_unpack, s.pack, s.unpack, s.unpack_from:
+            self.assertRaises(ValueError, meth, b'0')
+        self.assertRaises(ValueError, s.pack_into, bytearray(1), 0, b'0')
+        self.assertRaises(ValueError, s.__sizeof__)
```

Comment:
Also, realized this after submitting the approval. This is quite minor, and I still approve of the PR either way. Instead of using `s.__sizeof__`, I would recommend using `sys.getsizeof(s)`:
```suggestion
        self.assertRaises(ValueError, sys.getsizeof(s))
```

In general, it seems to be preferable to use functions instead of directly accessing the special object attributes when possible, and `sys.getsizeof()` performs automatic garbage collection when applicable.  If you had a specific reason for not using `sys.getsizeof()`, let me know. Here's the a link to the [function defintion](https://github.com/python/cpython/blob/59ad110d7a7784d53d0b502eebce0346597a6bef/Python/sysmodule.c#L1565) and the [docs](https://docs.python.org/3/library/sys.html#sys.getsizeof). I looked over it, but I'm not very experienced with the python c-api. I mostly rely on the docs when it comes to the modules implemented in c.